### PR TITLE
Harmonising domains for all ontologies

### LINKED ISSUE #1225: Categorisation scheme for OBO ontologies
To move forward on the recurring issues of grouping OBO ontologies into some useful macroscopic categories/domains, there were a few proposals:

- @zhengj2007 [Proposal](https://docs.google.com/spreadsheets/d/1pnsRsZluXcBlMqLfPmcCxXiiKxkr3s-nmuNve3kRMhE/edit#gid=78923188)
- My proposal (which I did not-knowing that @zhengj2007 had worked on one)
- @nataled ("using COB")

Aspects of this have been discussed [here](https://github.com/OBOFoundry/OBOFoundry.github.io/issues/19) (Portal pages), [here](https://github.com/OBOFoundry/OBOFoundry.github.io/issues/547) (Failed links to group pages) and [here](https://github.com/OBOFoundry/OBOFoundry.github.io/issues/965) (in the context of the revision of the "scope" principle). I would like to suggest this:

1. We have a small set (~10) of fixed domains to choose from.
2. Every ontology picks exactly one domain.
3. We have a more open-ended set of tags (optional field in metadata). Here, ontology devs can group their ontologies more deeply. Having this field will preclude any discussion about “I want more than one tag”.

My suggestion to start with is this and iterate over it together:

- upper
- anatomy and development
- health
- phenotype
- experiments
- environment
- biochemistry
- metabolomics and nutrition
- biological systems
- other

I looked at what values are currently used in the domain metadata field across OBO ontology metadata entries, and [sorted those into these proposed categories](https://docs.google.com/document/d/1fitsa5ZWKR_fidG0zC_eQIZDhmPXDpkG_oiql6nCDRo/edit).

Note that there are, of course, other possible categorisation schemes - starting from taxon (ontologies pertaining to plants, humans etc). Also @nataled point of simply using COB would be an option, but I think that domains that would be most useful are more like topics (Health, Biochemistry, Anatomy and Development etc) than concepts ("disease", "anatomical entity"), at least most of the time. I think that annotation with COB terms would be useful as well - maybe under a new element called `coverage`, that allows multiple elements. Also, the point here is to present some sensible high level groupings to the user, to help them to find relevant ontologies faster. I can see OLS picking this up as well.

### PR DESCRIPTION
**Pull request by matentzn**
Fixes #1225 

As we are finally starting to figure out the front page display, here the harmonised domain metadata for all ontologies according to https://github.com/OBOFoundry/OBOFoundry.github.io/issues/1225.

This PR

- Harmonises all ontologies to one of 15 domains, see #1225
- Moves all the previous domains into a different metadata element (tags)
- Constraints both the `tags` and the `domain` elements in the metadata schema, making `domain` a required enum.
- This manual "rough" categorisation is only for display purposes - once we have complete alignment with COB, we may be able to do something a bit cooler for finding ontologies, but this will likely by orthogonal.

### CODE REVIEW

File: util/schema/registry_schema.json [hunk 1]
```diff
@@ -96,10 +96,17 @@
       "suggest": false
     },
     "domain": {
-      "level": "warning",
-      "description": "'domain' is a recommended field. The value is the scientific topic area addressed by the ontology.",
+      "level": "error",
+      "description": "'domain' is a mandatory field. The value is the scientific topic area addressed by the ontology.",
```
Comment (cthoyt) [inline util/schema/registry_schema.json:100]: Consider converting to an enum of allowed values to make sure that no garbage sneaks in

Comment (cthoyt) [reply util/schema/registry_schema.json:100]: Ah you beat me to it!

Comment (matentzn) [reply util/schema/registry_schema.json:100]: Thanks done that now!

File: ontology/ecocore.md [hunk 2]
```diff
@@ -15,7 +15,9 @@ contact:
   github: pbuttigieg
   orcid: 0000-0002-4366-3088
 description: Ecocore is a community ontology for the concise and controlled description of ecological traits of organisms.
-domain: ecological functions, ecological interactions
+domain: environment
+tags:
+  - ecological functions, ecological interactions
```
Comment (cthoyt) [inline ontology/ecocore.md:20]: should be two separate tags?

File: ontology/eupath.md [hunk 3]
```diff
@@ -10,7 +10,9 @@ license:
   url: http://creativecommons.org/licenses/by/4.0/
   label: CC BY 4.0
 depicted_by: https://raw.githubusercontent.com/EuPathDB/communitysite/master/assets/images/VEuPathDB-logo-s.png
-domain: functional genomics, population biology, clinical epidemiology, and microbiomes
+domain: organisms
+tags:
+  - functional genomics, population biology, clinical epidemiology, and microbiomes
```
Comment (cthoyt) [inline ontology/eupath.md:15]: should be several separate tags?

File: ontology/oae.md [hunk 4]
```diff
@@ -7,7 +7,9 @@ contact:
   github: yongqunh
   orcid: 0000-0001-9189-9661
 description: A biomedical ontology in the domain of adverse events
-domain: adverse events, health
+domain: health
+tags:
+  - adverse events, health
```
Comment (cthoyt) [inline ontology/oae.md:12]: seems redundant of the domain
```suggestion
  - adverse events
```

File: ontology/obib.md [hunk 5]
```diff
@@ -8,7 +8,9 @@ contact:
   github: zhengj2007
   orcid: 0000-0002-2999-0103
 description: An ontology built for annotation and modeling of biobank repository and biobanking administration
-domain: biobanking, specimens, bio-repository, biocuration
+domain: experiments
+tags:
+  - biobanking, specimens, bio-repository, biocuration
```
Comment (cthoyt) [inline ontology/obib.md:13]: different tags? also I'd suggest removing "biocuration" since this is a bit redundant

File: ontology/opl.md [hunk 6]
```diff
@@ -10,7 +10,9 @@ license:
   url: https://creativecommons.org/licenses/by/4.0/
   label: CC BY 4.0
 description: A reference ontology for parasite life cycle stages.
-domain: life cycle stage, parasite organism
+domain: organisms
+tags:
+  - life cycle stage, parasite organism
```
Comment (cthoyt) [inline ontology/opl.md:15]: should be multiple tags

File: ontology/sepio.md [hunk 7]
```diff
@@ -3,7 +3,9 @@ layout: ontology_detail
 id: sepio
 title: Scientific Evidence and Provenance Information Ontology
 description: An ontology for representing the provenance of scientific claims and the evidence that supports them.
-domain: scientific claims, evidence
+domain: experiments
+tags:
+  - scientific claims, evidence
```
Comment (cthoyt) [inline ontology/sepio.md:8]: multiple tags

File: ontology/ecocore.md (hunk 2 above)
Comment (matentzn) [reply ontology/ecocore.md:20]: yeah could be

File: ontology/eupath.md (hunk 3 above)
Comment (matentzn) [reply ontology/eupath.md:15]: yeah could be

File: ontology/obib.md (hunk 5 above)
Comment (matentzn) [reply ontology/obib.md:13]: fine by me!

**Comment by matentzn** [comment]
I dont mind, but for now, i just want the domains to be amended and some kind of place where the previous information still lives. In fact, I would be even more inclined to drop the tags, but 1 step at a time. We need this merged like 1 year ago, because we have someone currently finally working on the front page :)

File: ontology/obib.md (hunk 5 above)
Comment (cthoyt) [reply ontology/obib.md:13]:
```suggestion
  - biobanking
  - specimens
  - bio-repository
```

File: ontology/eupath.md (hunk 3 above)
Comment (cthoyt) [reply ontology/eupath.md:15]:
```suggestion
  - functional genomics
  - population biology
  - clinical epidemiology
  - microbiomes
```

File: ontology/ecocore.md (hunk 2 above)
Comment (cthoyt) [reply ontology/ecocore.md:20]:
```suggestion
  - ecological functions
  - ecological interactions 
```

File: ontology/colao.md [hunk 8]
```diff
@@ -8,7 +8,9 @@ contact:
   github: JCGiron
   orcid: 0000-0002-0851-6883
 description: "The Coleoptera Anatomy Ontology contains terms used for describing the anatomy and phenotype of beetles in biodiversity research. It has been built using the Ontology Develoment Kit, with the Ontology for the Anatomy of the Insect Skeleto-Muscular system (AISM) as a backbone."
-domain: Insect Anatomy.
+domain: anatomy and development
+tags:
+  - Insect Anatomy.
```
Comment (cthoyt) [inline ontology/colao.md:13]: capitalization and punctuation don't match the style from the rest
```suggestion
  - insect anatomy
```

File: ontology/lepao.md [hunk 9]
```diff
@@ -8,7 +8,9 @@ contact:
   github: luis-gonzalez-m
   orcid: 0000-0002-9136-9932
 description: The Lepidoptera Anatomy Ontology contains terms used for describing the anatomy and phenotype of moths and butterflies in biodiversity research. LEPAO is developed in part by BIOfid (The Specialised Information Service Biodiversity Research).
-domain: Insect Anatomy
+domain: anatomy and development
+tags:
+  - Insect Anatomy
```
Comment (cthoyt) [inline ontology/lepao.md:13]:
```suggestion
  - insect anatomy
```

File: ontology/one.md [hunk 10]
```diff
@@ -9,7 +9,9 @@ contact:
   github: cyang0128
   orcid: 0000-0001-9202-5309
 description: An ontology to standardize research output of nutritional epidemiologic studies.
-domain: nutritional epidemiology, observational studies, dietary surveys
+domain: diet, metabolomics and nutrition
+tags:
+  - nutritional epidemiology, observational studies, dietary surveys
```
Comment (cthoyt) [inline ontology/one.md:14]:
```suggestion
  - nutritional epidemiology
  - observational studies
  - dietary surveys
```

File: ontology/ontoavida.md [hunk 11]
```diff
@@ -8,7 +8,9 @@ contact:
   github: miguelfortuna
   orcid: 0000-0002-8374-1941
 description: "OntoAvida develops an integrated vocabulary for the description of the most widely-used computational approach for studying evolution using digital organisms (i.e., self-replicating computer programs that evolve within a user-defined computational environment)."
-domain: digital evolution, a branch of Artificial Life.
+domain: other
+tags:
+  - digital evolution, a branch of Artificial Life.
```
Comment (cthoyt) [inline ontology/ontoavida.md:13]: are these two different tags?

File: ontology/pso.md [hunk 12]
```diff
@@ -15,7 +15,9 @@ contact:
   github: cooperl09
   orcid: 0000-0002-6379-8932
 description: The Plant Stress Ontology describes...
-domain: plant disease and abiotic stress
+domain: agriculture
+tags:
+  - plant disease and abiotic stress
```
Comment (cthoyt) [inline ontology/pso.md:20]: are these two different tags?

File: ontology/rbo.md [hunk 13]
```diff
@@ -15,7 +15,9 @@ contact:
   github: DanBerrios
   orcid: 0000-0003-4312-9552
 description: RBO is an ontology for the effects of radiation on biota in terrestrial and space environments.
-domain: radiation biology, the study of the effects of radiation on biological systems
+domain: environment
+tags:
+  - radiation biology, the study of the effects of radiation on biological systems
```
Comment (cthoyt) [inline ontology/rbo.md:20]: seems like this isn't the place for explanation
```suggestion
  - radiation biology
```

File: util/schema/registry_schema.json [hunk 14]
```diff
@@ -96,9 +96,33 @@
...
+        "upper",
+        "anatomy and development",
+        "health",
+        "phenotype",
+        "experiments",
+        "environment",
+        "biochemistry",
+        "microbiology",
+        "agriculture",
+        "diet, metabolomics and nutrition",
```
Comment (cthoyt) [inline util/schema/registry_schema.json:112]: would highly prefer this having an Oxford comma to reduce confusion later

File: ontology/pso.md (hunk 12 above)
Comment (matentzn) [reply ontology/pso.md:20]: Probably supposed to be a "conjunctive" expression, I think we can split this like the rest.

File: ontology/ontoavida.md (hunk 11 above)
Comment (matentzn) [reply ontology/ontoavida.md:13]: Why not. digital evolution. artificial life

File: util/schema/registry_schema.json (hunk 14 above)
Comment (matentzn) [reply util/schema/registry_schema.json:112]: If you can be bothered to make that fix, I think an oxford comma is appropriate.

File: util/schema/registry_schema.json [hunk 15]
```diff
@@ -96,9 +96,33 @@
...
+        "biological systems",
+        "information technology",
+        "organisms",
+        "information",
+        "other"
```
Comment (cthoyt) [inline util/schema/registry_schema.json:117]: the only "other" is the digital organisms one -> would be better to just reclassify as "organisms" then remove "other" so things don't get added without thinking this through very carefully

File: util/schema/registry_schema.json (hunk 14 above)
Comment (cthoyt) [reply util/schema/registry_schema.json:112]:
```suggestion
        "diet, metabolomics, and nutrition",
```

File: ontology/one.md [hunk 16]
```diff
@@ -9,7 +9,11 @@ contact:
   github: cyang0128
   orcid: 0000-0001-9202-5309
 description: An ontology to standardize research output of nutritional epidemiologic studies.
-domain: nutritional epidemiology, observational studies, dietary surveys
+domain: diet, metabolomics and nutrition
```
Comment (cthoyt) [inline ontology/one.md:12]:
```suggestion
domain: diet, metabolomics, and nutrition
```

File: ontology/ons.md [hunk 17]
```diff
@@ -9,7 +9,9 @@ contact:
   github: FrancescoVit
   orcid: 0000-0001-9125-4337
 description: An ontology for description of concepts in the nutritional studies domain.
-domain: nutrition, nutritional studies, nutrition professionals
+domain: diet, metabolomics and nutrition
```
Comment (cthoyt) [inline ontology/ons.md:12]:
```suggestion
domain: diet, metabolomics, and nutrition
```

File: util/schema/registry_schema.json (hunk 14 above)
Comment (cthoyt) [reply util/schema/registry_schema.json:112]: i added suggestions everywhere else too

File: ontology/ontoavida.md [hunk 18]
```diff
@@ -8,7 +8,9 @@ contact:
   github: miguelfortuna
   orcid: 0000-0002-8374-1941
 description: "OntoAvida develops an integrated vocabulary for the description of the most widely-used computational approach for studying evolution using digital organisms (i.e., self-replicating computer programs that evolve within a user-defined computational environment)."
-domain: digital evolution, a branch of Artificial Life.
+domain: other
```
Comment (cthoyt) [inline ontology/ontoavida.md:11]: Maybe a stretch, but still better than having an "other" category
```suggestion
domain: organisms
```

Comment (matentzn) [reply ontology/ontoavida.md:11]: hmm.. I would prefer a new category "artificial life" rather than forcing that into "organisms", but I can be outvoted.

Comment (alanruttenberg) [reply ontology/ontoavida.md:11]: Maybe simulation as a category. Includes cell simulations, for example, which we wouldn't call artificial life.

File: util/schema/registry_schema.json (hunk 15 above)
Comment (matentzn) [reply util/schema/registry_schema.json]: Replaced by new category "simulation" as @alanruttenberg suggested.

File: ontology/pso.md (hunk 12 above)
Comment (matentzn) [reply ontology/pso.md]: Split it!

Comment (matentzn) [reply ontology/pso.md]: I meant, I splitted it.

**Review by cthoyt — APPROVED** [review]
Thanks for considering all of the feedback, Nico. I'd suggest merging this now since it's at least 90% good enough, and OBO community discussions are prone to getting blocked by non-actionable feedback.

**Comment by matentzn** [comment]
Thank you @cthoyt for all the feedback.. I need to consider a bit the social side of this, but I will merge this soon, given how much this topic has been discussed. This PR will be merged by the Wizard of Oz on the evening of the 11th March.

**Comment by cthoyt** [comment]
@matentzn time to merge ;)

**Pull request merged**